storage/blockdevice/source/SFDP.cpp

```diff
@@ -291,13 +291,96 @@ int sfdp_parse_sector_map_table(Callback<int(bd_addr_t, sfdp_cmd_addr_size_t, ui
         return -1;
     }
 
-    // Currently we support only Single Map Descriptor
-    if (!((smptbl_buff[0] & 0x3) == 0x03) && (smptbl_buff[1] == 0x0)) {
-        tr_error("Sector Map: Supporting Only Single Map Descriptor (not map commands)");
+    // Loop through all descriptors: configuration command descriptor and sector map descriptors.
+    // Command descriptors are optional. If any exist, they always precede sector map descriptors.
+    uint8_t *table = smptbl_buff.get();
+    uint8_t *descriptor = table;
+    const size_t min_descriptor_size = 8; // two DWORDs
+
+    // Part 1: configuration detection command descriptors (optional).
+    // If any exist, run their commands to determine the active configuration's ID.
+    // Otherwise the configuration ID is 0x00.
+    uint8_t active_config_id = 0x00;
+    // If any configuration detection command descriptors exist, the table starts with one.
+    if (!(descriptor[0] & 0x02)) {
```

Comment:
Should `0x01` and `0x02` be named variables? We use them more than once in this code.

Reply:
I'll name those if they are used multiple times. I tried not to alias any bit masks that are used once only, as the code should be self-explanatory (i.e. what they are for).